File: mobile_ui/quickchat_ui/lib/features/authentication/usecase/register_user.usecase.dart
import 'package:injectable/injectable.dart';
import 'package:quickchat_ui/features/authentication/data/bloc/auth_bloc.dart';
import 'package:quickchat_ui/features/authentication/data/dto/RegisterUser.dto.dart';

@injectable
class RegisterUserUsecase {
  RegisterUserUsecase({required this.authBloc});

  final AuthBloc authBloc;

  void execute(RegisterUserDto req) {
    authBloc.add(AuthEvent.registerUser(registerUserDto: req));
  }
}

File: web_ui/app/features/authentication/domain/usecases/registerUser.usecase.ts
import { RegisterUserDto } from "../../data/dto/RegisterUser.dto";
import { registerUser } from "../../data/redux/authSlice";

export class RegisterUserUsecase {
  async execute(
    {
      registerUserDto,
    }: {
      registerUserDto: RegisterUserDto;
    },
    dispatch: any
  ): Promise<void> {
    dispatch(registerUser(registerUserDto));
  }
}
